File: openmrs-client/src/main/java/org/openmrs/mobile/models/Session.java
/*
 * The contents of this file are subject to the OpenMRS Public License
 * Version 1.0 (the "License"); you may not use this file except in
 * compliance with the License. You may obtain a copy of the License at
 * http://license.openmrs.org
 *
 * Software distributed under the License is distributed on an "AS IS"
 * basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See the
 * License for the specific language governing rights and limitations
 * under the License.
 *
 * Copyright (C) OpenMRS, LLC.  All Rights Reserved.
 */

package org.openmrs.mobile.models;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Session {

    @SerializedName("sessionId")
    @Expose
    private String sessionId;
    @SerializedName("authenticated")
    @Expose
    private boolean authenticated;
    @SerializedName("user")
    @Expose
    private User user;

    public Session(String sessionId, boolean authenticated, User user) {
        this.sessionId = sessionId;
        this.authenticated = authenticated;
        this.user = user;
    }

    public String getSessionId() {
        return sessionId;
    }

    public void setSessionId(String sessionId) {
        this.sessionId = sessionId;
    }

    public boolean isAuthenticated() {
        return authenticated;
    }

    public void setAuthenticated(boolean authenticated) {
        this.authenticated = authenticated;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    @Override
    public String toString() {
        return "\tsessionId: " + sessionId + "\tauthenticated: " + authenticated;
    }
}

File: openmrs-android-sdk/src/main/java/com/example/openmrs_android_sdk/library/models/Session.kt
/*
 * The contents of this file are subject to the OpenMRS Public License
 * Version 1.0 (the "License"); you may not use this file except in
 * compliance with the License. You may obtain a copy of the License at
 * http://license.openmrs.org
 *
 * Software distributed under the License is distributed on an "AS IS"
 * basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See the
 * License for the specific language governing rights and limitations
 * under the License.
 *
 * Copyright (C) OpenMRS, LLC.  All Rights Reserved.
 */

package com.example.openmrs_android_sdk.library.models

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class Session(@field:SerializedName("sessionId")
              @field:Expose
              var sessionId: String?, @field:SerializedName("authenticated")
              @field:Expose
              var isAuthenticated: Boolean, @field:SerializedName("user")
              @field:Expose
              var user: User?) {

    override fun toString(): String {
        return "\tsessionId: $sessionId\tauthenticated: $isAuthenticated"
    }
}
